src/plans/conjugate_gradient_plan.jl

```diff
@@ -67,7 +86,7 @@ function ConjugateGradientDescentState(
     p::P,
     sC::StoppingCriterion,
     s::Stepsize,
-    dU::DirectionUpdateRule,
+    dU::DirectionUpdateRuleStorage,
```

Comment:
since you have functions that automatically „know“ the storage requirements, we could keep the constructor nonbreaking and internally create the new storage rule?

Reply:
Maybe even a stronger argument here is that the coefficient can not be used alone _and_ the storage is an internal thing we do, so we should keep that also more internal.

Reply:
Good idea, I've reverted this change.